### PR TITLE
tx-pool: Make txpool independent of primitive tx types

### PR DESCRIPTION
**Pull request by tcoratger**
Related #8668

### CODE REVIEW

**Comment by JackG-eth** [comment]
@tcoratger Just want to make sure i'm not blocking you here or anything? looks like its a finished version of mine!

**Comment by tcoratger** [comment]
> @tcoratger Just want to make sure i'm not blocking you here or anything? looks like its a finished version of mine!

@JackG-eth I think this will be reviewed soon by @mattsse but this is just scaffolding as the issue was "quite urgent" but this doesn't close completely the issue as we still have the constraints:

```rust
/// Trait for transaction types used inside the pool
pub trait PoolTransaction:
    fmt::Debug
    + Send
    + Sync
    + Clone
    + From<PooledTransactionsElementEcRecovered>
    + TryFrom<TransactionSignedEcRecovered>
    + Into<TransactionSignedEcRecovered>
{
```

with

```rust
    + From<PooledTransactionsElementEcRecovered>
    + TryFrom<TransactionSignedEcRecovered>
    + Into<TransactionSignedEcRecovered>
```

that should be removed completely in the abstracted version to close the issue. Indeed, in the end what we want is to no longer depend in this trait on specific type of transactions and this is not yet the case here. I just made simplifications and put the necessary types in place to facilitate the work later as doing everything on a single PR would have been too long and time consuming to review.


What I think you can do is once this PR is merged you can go back from there to do the final abstraction and finish the job, wdyt @mattsse ?

**Comment by mattsse** [comment]
great, this sgtm!

reviewing in a bit, ty!

File: crates/transaction-pool/src/traits.rs [hunk 1]
```diff
@@ -758,10 +757,17 @@ pub trait PoolTransaction:
...
+    + Clone
+    + From<PooledTransactionsElementEcRecovered>
+    + TryFrom<TransactionSignedEcRecovered>
+    + Into<TransactionSignedEcRecovered>
 {
+    /// Associated type representing the raw consensus variant of the transaction.
+    type Consensus: From<Self> + TryInto<Self>;
+
+    /// Associated type representing the recovered pooled variant of the transaction.
+    type Pooled: Into<Self>;
```
Comment (mattsse) [inline crates/transaction-pool/src/traits.rs:769]: this makes sense as a low invasive step towards that feature, nice

**Review by mattsse — APPROVED** [review]
clean, ty for doing this incrementally

pending conflict

**Review by mattsse — APPROVED** [review]

**Pull request merged**